From cae38feb629c7d30c53366e5b48e96b759b7963d Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Tue, 4 Apr 2017 16:23:42 +0200
Subject: [PATCH] tc: m_xt: Get rid of iargc variable in parse_ipt()

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1326726
Upstream Status: iproute2.git commit 28432f370e6ad

commit 28432f370e6ad72a7eab9ba6bae67eabdd4b1c57
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Jun 10 13:42:06 2016 +0200

    tc: m_xt: Get rid of iargc variable in parse_ipt()

    After dropping the unused decrement of argc in the function's tail, it
    can fully take over what iargc has been used for.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 tc/m_xt.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/tc/m_xt.c b/tc/m_xt.c
index 7947e5a..e0ed8a4 100644
--- a/tc/m_xt.c
+++ b/tc/m_xt.c
@@ -124,7 +124,7 @@ static int parse_ipt(struct action_util *a,int *argc_p,
 
 	int c;
 	char **argv = *argv_p;
-	int argc = 0, iargc = 0;
+	int argc = 0;
 	char k[16];
 	int size = 0;
 	int iok = 0, ok = 0;
@@ -144,7 +144,7 @@ static int parse_ipt(struct action_util *a,int *argc_p,
 				break;
 			}
 		}
-		iargc = argc = i;
+		argc = i;
 	}
 
 	if (argc <= 2) {
@@ -205,7 +205,7 @@ static int parse_ipt(struct action_util *a,int *argc_p,
 		}
 	}
 
-	if (iargc > optind) {
+	if (argc > optind) {
 		if (matches(argv[optind], "index") == 0) {
 			if (get_u32(&index, argv[optind + 1], 10)) {
 				fprintf(stderr, "Illegal \"index\"\n");
@@ -270,9 +270,8 @@ static int parse_ipt(struct action_util *a,int *argc_p,
 		addattr_l(n, MAX_MSG, TCA_IPT_TARG, m->t, m->t->u.target_size);
 	tail->rta_len = (void *) NLMSG_TAIL(n) - (void *) tail;
 
-	argc -= optind;
 	argv += optind;
-	*argc_p -= iargc;
+	*argc_p -= argc;
 	*argv_p = argv;
 
 	optind = 0;
-- 
1.8.3.1

